src/main/java/athena/TimeAllocator.java

```diff
@@ -106,6 +107,8 @@ private void assignTime(ArrayList<Integer> bestLog, int pos, LocalDate currDay)
                     this.flexibleTaskList.deleteTask(taskNumber);
                 } catch (TaskNotFoundException e) {
                     //do nothing
+                } catch (DateHasPassedException e) {
+                    assert false; // Does this make sense?
```

Comment:
Need to make sure Storage doesn't load in old tasks. I suppose it shouldn't be an issue because it looks like you're working on that.

In the hypothetical situation when the user leaves ATHENA running for more than a day then this might crash lol.

Reply:
I think it might make more sense to self destruct tasks when you catch this error, but then we will have to have another self destruct for repeating tasks